src/HttpClient.php

```diff
@@ -91,6 +93,67 @@ public function deploy(array $data)
         $this->guzzle->post('deploy', ['json' => $data]);
     }
 
+    /**
+     * Notify Bugsnag of a build.
+     *
+     * @param array $data the build information
+     *
+     * @return void
+     */
+    public function sendBuildReport(array $buildInfo)
+    {
+        $app = $this->config->getAppData();
+
+        $data = [];
+        $sourceControl = [];
+        if (isset($buildInfo['repository'])) {
+            $sourceControl['repository'] = $buildInfo['repository'];
+        }
+
+        if (isset($buildInfo['provider'])) {
+            $sourceControl['provider'] = $buildInfo['provider'];
+        } elseif (isset($buildInfo['repository'])) {
+            $url = $buildInfo['repository'];
+            if (strpos($url, 'github.com') != false) {
+                $sourceControl['provider'] = 'github';
+            } elseif (strpos($url, 'bitbucket.com') != false) {
+                $sourceControl['provider'] = 'bitbucket';
+            } elseif (strpos($url, 'gitlab.com')) {
+                $sourceControl['provider'] = 'gitlab';
+            }
+        }
+
+        if (isset($buildInfo['revision'])) {
+            $sourceControl['revision'] = $buildInfo['revision'];
+        }
+
+        if (!empty($sourceControl)) {
+            $data['sourceControl'] = $sourceControl;
+        }
+
+        if (isset($buildInfo['builder'])) {
+            $data['builderName'] = $buildInfo['builder'];
+        }
+
+        if (isset($buildInfo['buildTool'])) {
+            $data['buildTool'] = $buildInfo['buildTool'];
```

Comment:
You shouldnt be able to externally set this

Reply:
Its settable for laravel to override

Reply:
I dont think we care about php/laravel distinction at an analytics level. We just care that this lib is being used for deploys.